### ISSUE TITLE
Balancing of AI attack and defense

### ISSUE THREAD

**Issue by FireWolf74** [endless-sky/endless-sky#3812]
When is the balancing of the AI for both attack and defense going to happen? Just for giggle today I took and gave myself negative rep for the Republic and then less than 2 mins a Republic Carrier (1) jumped into a sector and deployed 9 drones and fighter and swarmed to kill quickly.  That is way to fast of a kill ratio, I understand the swarming aspect but I can't deploy fighters from my Korath Raider that quickly.

I was bored waiting for the next update.

**Comment by KiLEdEnNis** [comment]
I'm sorry, what?

**Comment by FireWolf74** [comment]
Let me see if I can break this down in to Machine Learning for everyone here.

When an AI attacks it runs pre programmed set of 1 and 0. When your AI attacks it creates an On (1) instead of an Off (0), now the way your AI attack is to send a bunch of fighters, drones and anything else it may have in the fleet or ship to attack a enemy that you set a value at.  The issue is the following the swarming of any target by a bunch of gnats instead of key learning in attacking certain targets of value.  
Case in point on the FW mission when retaking certain planets the value of a Player is giving a higher value (the swarm targets) instead of all the ships within that attack group being the same.  Certain ships should be given a certain value higher the ship the higher the target.  Your AI is not factoring this into fact, you told it that the Player has the highest value because it is not an NPC, thus the players are targeted and killed at a higher ratio than NPCs.  

Let me see if I can break it down further,  KiLEdEnNis myself and you are in a Military unit your the commander and I'm under you.  You given me a task to take out a certain HVT (High Value Target) but you tell me I must follow the plan you lay out , following me so far right?  Now with me I have 5 other people and they are working together with me to take out the HVT but upon starting the mission I'm killed by say 50 bow and arrows while the 5 other people are running around shooting things.  

If this is too hard to follow I'm sorry but I'm using my Military mind which sees all sides.  Also I understand from a professional level that your trying to make it fun for everyone but balancing this is just as important as other things you have in the works.
Thoughts from a military mind.
2/11 ACR ALLONS

**Comment by KiLEdEnNis** [comment]
so you're saying hostile ships target the player ship too much? unless they have the tag "nemesis" they usually pick the closest ship to them, so it's not really an AI problem
if you keep dying in fleet combat like an FW vs Navy battle, try to stick with bigger friendly ships and target the same ship as them, or get a better warship

was that your problem?

**Comment by temtemy** [comment]
I guess I have not made it clear to you in https://github.com/endless-sky/endless-sky/issues/3786#issuecomment-397933296:

> Endless Sky is a FLOSS (free, libre, and open source) game. There is no precise time when a feature will be added.

Endless Sky is developed by unpaid volunteers. We help the game's development in our small spare time. All of us have a life. Because of that, no one knows when the heck something will be done. Look, I have been waiting for months when 0.9.9 will be released. Did I opened an issue and asked? No.

I think it is normal for the AI to deploy the fighters and drones once a hostile appeared. It is a good tactic actually, since you can distract your target, and of course, more damage to the target. Of course, the fighters and drones will die quickly anyway, but they did their duty for honor. Even some Navy soldiers acknowledge that fighter duty is suicide. So, TLDR, no problem at all.

Please close this issue. Or better, try to "balance" the AI yourself and open a PR. That's much appreciated.

**Comment by temtemy** [comment]
Also, regarding the Raider thing, I noticed that too, it deploys its fighters slower than the human carriers. If that's the real issue, then rename your issue and probably others will be interested in discussing that with you.

**Comment by FireWolf74** [comment]
Temtemy, 
"I think it is normal for the AI to deploy the fighters and drones once a hostile appeared. It is a good tactic actually, since you can distract your target, and of course, more damage to the target. Of course, the fighters and drones will die quickly anyway, but they did their duty for honor. Even some Navy soldiers acknowledge that fighter duty is suicide." 
I guess I put value on the Soliders, and Sailors I have in my ships.
Each Solider makes a unit better. Leaders are nothing without Soliders, Sailors, and Marines.

**Comment by jafdy** [comment]
A stock Carrier with stock fighter and drones against a Raider without and shield generation and without fighting back in any way will take at least 12 seconds in the worst case to drop the shields of the Raider. The Raider will last quite a bit longer than that because of the hull and shield generation/repair and because some of the fighters will be destroyed. Other Carrier variants (excluding the alpha Carrier) will not be much more powerful. 
Unless your reactions measure in the tens of seconds you shouldn't have too much of an issue winning or at least running away from this fight.
